Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt

```diff
@@ -102,20 +108,46 @@ class DetailsActivity : ComponentActivity() {
     }
 }
 
+private data class DetailsScreenUiState(
+    val exploreModel: ExploreModel? = null,
+    val isLoading: Boolean = false,
+    val throwError: Boolean = false
+)
+
 @Composable
 fun DetailsScreen(
     onErrorLoading: () -> Unit,
     modifier: Modifier = Modifier,
     viewModel: DetailsViewModel = viewModel()
 ) {
-    val cityDetailsResult = remember(viewModel) { viewModel.cityDetails }
-    if (cityDetailsResult is Result.Success<ExploreModel>) {
-        DetailsContent(cityDetailsResult.data, modifier)
+    val uiState by produceState(
+        key1 = viewModel,
+        initialValue = DetailsScreenUiState(isLoading = true)
+    ) {
+        val cityDetailsResult = viewModel.cityDetails
+        value = if (cityDetailsResult is Result.Success<ExploreModel>) {
+            DetailsScreenUiState(cityDetailsResult.data)
```

Comment:
Should the mapping to `DetailsScreenUiState` be done in the VM?

Reply:
Added comment explaining why. Thanks!